Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -113,6 +124,28 @@ internal class MessagingPushImplementation: MessagingPushInstance {
                                     data: MetricRequest(deliveryId: deliveryID, event: event, deviceToken: deviceToken,
                                                         timestamp: Date()))
     }
+    func deviceAttributes(deviceToken: String, completionHandler: @escaping([String: String]?) -> Void) {
+        if !sdkConfigStore.config.autoTrackDeviceAttributes {
+            completionHandler(nil)
+            return
+        }
+        #if canImport(UIKit)
```

Comment:
The `completionHandler` is never called if `#if canImport(UIKit)` is false. 

Read my comment for `DeviceInfo` enum. You may not need this `#if canImport()` at all.

Reply:
👍 ✅